Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/Sos.hs

```diff
@@ -50,26 +101,119 @@ newtype SosRes = SosRes
+            customerPhoneNo = phoneNumber,
+            driverName = Just ride.driverName,
+            driverPhoneNo = Just ride.driverMobileNumber,
+            vehicleNo = ride.vehicleNumber,
+            status = show ride.status,
+            rideCreatedAt = ride.createdAt,
+            pickupLocation = castLocationAPIEntity ride.fromLocation,
+            dropLocation = castLocationAPIEntity <$> ride.toLocation,
+            fare = Nothing
+          }
+      trackLink = merchantConfig.trackingShortUrlPattern <> ride.shortId.getShortId
+  ticketId <- do
+    if merchantConfig.enableSupportForSafety
+      then do
+        ticketResponse <- try @_ @SomeException (createTicket person.merchantId person.merchantOperatingCityId (mkTicket person phoneNumber trackLink rideInfo))
+        case ticketResponse of
+          Right ticketResponse' -> return (Just ticketResponse'.ticketId)
+          Left _ -> return Nothing
+      else return Nothing
+  when (person.shareEmergencyContacts) $ do
+    emergencyContacts <- getDefaultEmergencyNumbers (personId, merchantId)
+    let sender = smsCfg.sender
+    message <-
+      MessageBuilder.buildSOSAlertMessage person.merchantOperatingCityId $
+        MessageBuilder.BuildSOSAlertMessageReq
+          { userName = (fromMaybe "" person.firstName) <> " " <> (fromMaybe "" person.lastName),
+            rideLink = trackLink
+          }
+    for_ emergencyContacts.defaultEmergencyNumbers $ \emergencyContact -> do
+      let emergencyPhoneNumber = emergencyContact.mobileCountryCode <> emergencyContact.mobileNumber
+      Sms.sendSMS person.merchantId person.merchantOperatingCityId (Sms.SendSMSReq message emergencyPhoneNumber sender)
+        >>= Sms.checkSmsResult
+  sosDetails <- buildSosDetails personId req ticketId
   _ <- QSos.create sosDetails
+
   return $
     SosRes
       { sosId = sosDetails.id
       }
+  where
+    castLocationAPIEntity ent =
+      Ticket.Location
+        { lat = ent.lat,
+          lon = ent.lon,
+          street = ent.address.street,
+          city = ent.address.city,
+          state = ent.address.state,
+          country = ent.address.country,
+          building = ent.address.building,
+          areaCode = ent.address.areaCode,
+          area = ent.address.area
+        }
 
-updateSosDetails :: (CacheFlow m r, EsqDBReplicaFlow m r, EsqDBFlow m r, EncFlow m r) => Id DSos.Sos -> Id Person.Person -> SosFeedbackReq -> m APISuccess.APISuccess
-updateSosDetails sosId personId req = do
-  sosDetails <- runInReplica $ QSos.findById sosId >>= fromMaybeM (SosIdDoesNotExist sosId.getId)
+getSosDetails :: (CacheFlow m r, MonadFlow m, EsqDBFlow m r) => Id DRide.Ride -> Id Person.Person -> m SosDetailsRes
+getSosDetails rideId_ personId_ = do
+  sosDetails <- QSos.findByRideIdAndStatus rideId_ DSos.Resolved
+  unless (personId_ == maybe "" (.personId) sosDetails) $ throwError $ InvalidRequest "PersonId not same"
+  case sosDetails of
```

Comment:
You can use  
`SosDetailsRes 
 {  sosId =  (.id) <$> sosDetails }`
 rather than using 
  ```
 case sosDetails of
     Just DSos.Sos {..} ->
      return $
        SosDetailsRes
          { sosId = Just id
          }
    Nothing ->
      return $
        SosDetailsRes
          { sosId = Nothing
          }
```